Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -284,17 +284,50 @@ public BlackboardArtifactNode(BlackboardArtifact artifact) {
      */
     private static Lookup createLookup(BlackboardArtifact artifact) {
         final long objectID = artifact.getObjectID();
+        Content content = null;
         try {
-            Content content = contentCache.get(objectID, () -> artifact.getSleuthkitCase().getContentById(objectID));
+            if (artifact.getArtifactTypeID() == BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_DOWNLOAD.getTypeID() || artifact.getArtifactTypeID() == BlackboardArtifact.ARTIFACT_TYPE.TSK_WEB_CACHE.getTypeID()) {
+                content = getPathIdFile(artifact);
+            }
             if (content == null) {
-                return Lookups.fixed(artifact);
-            } else {
-                return Lookups.fixed(artifact, content);
+                content = contentCache.get(objectID, () -> artifact.getSleuthkitCase().getContentById(objectID));
             }
         } catch (ExecutionException ex) {
             logger.log(Level.SEVERE, MessageFormat.format("Error getting source content (artifact objID={0}", artifact.getId()), ex); //NON-NLS
+            content = null;
+        }
+        if (content == null) {
             return Lookups.fixed(artifact);
+        } else {
+            return Lookups.fixed(artifact, content);
+        }
+
+    }
+
+    /**
+     * Private helper method to allow content specified in a path id attribute
+     * to be retrieved.
+     *
+     * @param artifact The artifact which content may be specified as a tsk path
+     *                 attribute.
+     *
+     * @return The Content specified by the artifact's path id attribute or null
+     *         if there was no content available.
+     *
+     * @throws ExecutionException Error retrieving the file specified by the
+     *                            path id from the cache.
+     */
+    private static Content getPathIdFile(BlackboardArtifact artifact) throws ExecutionException {
+        try {
+            for (BlackboardAttribute attribute : artifact.getAttributes()) {
+                if (attribute.getAttributeType().getTypeID() == BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PATH_ID.getTypeID()) {
+                    return contentCache.get(attribute.getValueLong(), () -> artifact.getSleuthkitCase().getContentById(attribute.getValueLong()));
+                }
+            }
+        } catch (TskCoreException ex) {
+            logger.log(Level.WARNING, MessageFormat.format("Error getting content for path id attrbiute for artifact: ", artifact.getId()), ex); //NON-NLS
```

Comment:
I think that we can simplify this code slightly by taking advantage of this BlackboardArtifact method:
BlackboardAttribute getAttribute(BlackboardAttribute.Type attributeType).

Reply:
Good catch. Thank you.